modules/core/shared/src/main/scala-3/dumbo/ResourceFilePath.scala

```diff
@@ -23,21 +23,22 @@ object ResourceFilePath:
       case head :: Nil =>
         val base = Paths.get(head.toURI())
         val resources =
-          new File(base.toString()).list().map(fileName => fromPath(Path("/") / location / fileName)).toList
+          new File(base.toString()).list().map(fileName => s"/$location/$fileName").toList
         Expr(resources)
       case Nil => report.errorAndAbort(s"resource ${location} was not found")
       case multiple =>
         report.errorAndAbort(s"found multiple resource locations for ${location} in:\n${multiple.mkString("\n")}")
 
-  private def applyImpl(x: Expr[String])(using Quotes): Expr[ResourceFilePath] =
+  private def fromResource(x: Expr[String])(using Quotes): Expr[ResourceFilePath] =
     import quotes.reflect.report
     val location = x.valueOrAbort
     if getClass().getResourceAsStream(location) != null then x
     else report.errorAndAbort(s"resource ${location} was not found")
 
-  inline def apply(name: String): ResourceFilePath       = ${ applyImpl('name) }
-  inline private def fromPath(p: Path): ResourceFilePath = p.toString()
+  inline def fromResource(name: String): ResourceFilePath = ${ fromResource('name) }
+  def apply(name: String): ResourceFilePath               = name
 
   extension (s: ResourceFilePath)
-    inline def value: String    = s
-    inline def toNioPath: JPath = Paths.get(s)
+    inline def value: String                       = s
+    inline def append(p: String): ResourceFilePath = s + p
+    inline def fileName: String                    = Path(s).fileName.toString
```

Comment:
In Scala 3 `ResourceFilePath.fromResourcesDir` builds a `List[ResourceFilePath]` at compile time using the new Scala 3 "macro magic" :D an F[_] would be an overkill here (if even possible). This way it works with Scala native as well as we can't list embedded resources at runtime on native. I got the initial idea from the implementation in [indoorvivants/roach](https://github.com/indoorvivants/roach/blob/main/module-core/src/main/scala/roach/ResourceFile.scala) (thanks to @keynmol btw. :raised_hands:    ). 

In Scala 2 I just didn't bother to use macros, it looked more complicated to me, but is probably possible as well. Instead it will read the resources that ship embedded within a jar file at runtime, that's why it's wrapped in an F[_]. Works on the JVM only though.

Not sure how many people would like to use Scala 2 with Scala Native, if there is going to be demand one could try to implement it the same way as with Scala 3 using macros.